### ISSUE TITLE
[Popover][Tooltip][HoverCard][Menu] Access to the runtime `data-side` value for JS animation

### ISSUE THREAD

**Issue by jjenzz** [radix-ui/primitives#791]
If using a JS lib for animation, the runtime value of `data-side` is not available to these libs so they cannot apply origin-aware animations.

This is a [sandbox provided in our discord](https://codesandbox.io/s/prod-snowflake-z4g2z?file=/App.js) where someone has attempted to use the `data-side` attribute as a prop by utilising our `Slot` component. Unfortunately, the attribute value will not be available when the animation component mounts so the animation never plays out.

I'm not sure how we can solve this given that we need the content to render for a frame in order to work out what side it should appear but I'm raising this ticket for us to have a think.

**Comment by benoitgrelard** [comment]
Yeah, that might be quite tricky indeed! We would have to think about this quite a bit, maybe some kind of hook or render prop could help.

**Comment by jjenzz** [comment]
Perhaps an `onSideChange` callback so they can control it and then our current `side` prop would become `defaultSide` which suits what it is actually doing more accurately.

**Comment by Lexachoc** [comment]
Any update on this?

Also, there's `[data-side]` for [Select](https://www.radix-ui.com/primitives/docs/components/select#content). Will Select also be considered to have the callback on side changes?

I saw there's a PR to expose `data-side` and `data-align` but for Popover anchor.

**Comment by 3pleFly** [comment]
I ended up using a mutation observer too, but it would be really nice to have this :)

**Comment by chaance** [comment]
No plans to implement at the moment, but we may revisit in the future.
